### PR TITLE
feat: Persist expand collapse

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Persists state of expanded accordions for configure and admin + option for more.
Screenshot 2024-05-27 at 15 13 27

Next PRs: I will create another PR with refactoring for the growing navigation sidebar components and unit tests for the navigation sidebar.

### CODE REVIEW

File: frontend/src/component/layout/MainLayout/NavigationSidebar/NavigationSidebar.tsx [hunk 1]
```diff
@@ -297,8 +298,10 @@ const useShowBadge = () => {
     return showBadge;
 };
 
+type NavigationMode = 'mini' | 'full';
```
Comment (kwasniew) [inline frontend/src/component/layout/MainLayout/NavigationSidebar/NavigationSidebar.tsx:301]: extract type refactoring

File: frontend/src/component/layout/MainLayout/NavigationSidebar/NavigationSidebar.tsx [hunk 2]
```diff
@@ -443,15 +446,40 @@ export const MobileNavigationSidebar: FC<{ onClick: () => void }> = ({
     );
 };
 
+const useExpanded = <T extends string>() => {
```
Comment (kwasniew) [inline frontend/src/component/layout/MainLayout/NavigationSidebar/NavigationSidebar.tsx:449]: new hook for storing the expansion settings

File: frontend/src/utils/unique.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export const unique = <T extends string | number>(items: T[]): T[] =>
```
Comment (kwasniew) [inline frontend/src/utils/unique.ts:1]: copied from the backend code

**Review by FredrikOseberg — APPROVED** [review]
LGTM

**Pull request merged**